File: frontend/src/types/workbench.ts
export interface WorkbenchCounts {
  participated_tasks: number
  competition_tasks: number
  pending_task_reviews: number
  pending_competition_reviews: number
}

export interface ClosureStatistics {
  participations: number
  submissions: number
  accepted_submissions: number
  competition_task_completions: number
  published_results: number
}

export interface WorkbenchResponse {
  counts: WorkbenchCounts
  statistics: ClosureStatistics
}

File: backend/app/schemas/workbench.py
"""Personal workbench counts and recomputable closure statistics."""

from __future__ import annotations

from pydantic import BaseModel


class WorkbenchCounts(BaseModel):
    participated_tasks: int
    competition_tasks: int
    pending_task_reviews: int
    pending_competition_reviews: int


class ClosureStatistics(BaseModel):
    participations: int
    submissions: int
    accepted_submissions: int
    competition_task_completions: int
    published_results: int


class WorkbenchResponse(BaseModel):
    counts: WorkbenchCounts
    statistics: ClosureStatistics
